crates/bitcoin-da/src/spec/header.rs

```diff
@@ -1,33 +1,39 @@
-use bitcoin::block::Header;
-use bitcoin::hash_types::{TxMerkleNode, WitnessMerkleNode};
+use bitcoin::block::Header as BitHeader;
+use bitcoin::hash_types::WitnessMerkleNode;
+use bitcoin::hashes::Hash;
 use bitcoin::BlockHash;
+use borsh::{BorshDeserialize, BorshSerialize};
 use serde::{Deserialize, Serialize};
 use sov_rollup_interface::da::BlockHeaderTrait;
 
 use super::block_hash::BlockHashWrapper;
 
-// BlockHashWrapper is a wrapper around BlockHash to implement BlockHashTrait
-#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+// HeaderWrapper is a wrapper around BlockHash to implement BlockHeaderTrait
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, BorshDeserialize, BorshSerialize, Serialize, Deserialize,
+)]
 pub struct HeaderWrapper {
-    header: Header, // not pub to prevent uses like block.header.header.merkle_root
+    header: OurHeader, // not pub to prevent uses like block.header.header.merkle_root
```

Comment:
```suggestion
    header: Header, // not pub to prevent uses like block.header.header.merkle_root
```
The name is free since you import `Header` from bitcoin crate as `BitHeader` which i would say should be `BitcoinHeader` no need to shorten that one.